ls1intum/Artemis
# [Bugfix] Correctly unlock programming exercises for exam

### Checklist

### Motivation and Context


Closes #3663 

### Description

This PR changes the behaviour of the scheduling to also schedule exercises from exams that are created right before the release.

### Steps for Testing


1. Log in to Artemis
2. Create an exam with a Release date that is 15-20 minutes in the future
2.1. Create a programming exercise in that exam
2.2. Sign up some students for the exam
2.3. Create the student exams and prepare the exercises (This step has to be finished at least 6 minutes before the release date)
2.4. Check that the students have access to their repositories during the exam (see below on how to check that)

3. Create another exam and repeat the steps, but do steps 2.1 to 2.3 during the last 5 minutes before the release of the exam.
3.1 Check that the students have access to their repositories during the exam (see below on how to check that)

How to check the student access:
- Either clone the repository as a student during the exam
- Or take a look at bitbucket. Select the repository of one student and check in the settings under "Repository-Berechtigungen"

### Test Coverage
